File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        map<int,int> hashMap; 
        int longest = 0; 
        for(int n : nums) hashMap[n]++;
        for(int i = 0; i < nums.size(); i++){
            int curr = nums[i]; 
            if(!hashMap.count(curr-1)){
                int streak = 0; 
                while(hashMap.count(curr) && hashMap[curr] > 0){
                    hashMap[curr]--; 
                    streak++; 
                    curr = curr + 1; 
                }
                longest = max(longest, streak); 
            }
        }
        return longest; 
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Map<Integer,Integer> hashMap = new HashMap<>(); 
        for(int n : nums) hashMap.put(n, hashMap.getOrDefault(n,0) + 1); 
        int max_ = 0; 
        for(int n : nums){
            int curr = n; 
            if(!hashMap.containsKey(curr-1)){
                int streak = 0; 
                while(hashMap.containsKey(curr) && hashMap.get(curr) > 0){
                    hashMap.put(curr, hashMap.get(curr) -1); 
                    streak++; 
                    curr++; 
                }
                max_ = Math.max(max_, streak); 
            }
        }

        return max_; 
    }
}